### ISSUE TITLE
[Bug]: Electron 32.1 does not use the portals inside a flatpak application

### ISSUE THREAD

**Issue by bermeitinger-b** [electron/electron#43819]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.1.0

### What operating system(s) are you using?

Other Linux

### Operating System Version

Archlinux (and others)

### What arch are you using?

x64

### Last Known Working Electron version

32.0.2

### Expected Behavior

Selecting a file or folder should use the GTK portals to read and write

### Actual Behavior

The GTK portal is not used but the generic filechooser inside the flatpak application is opened without read/write access to any file on the host.

### Additional Information

This came up in https://github.com/flathub/org.signal.Signal/issues/719 where we found that starting from Electron 32, the portals are not used anymore.

Sadly, I'm not skilled enough with Electron, packaging, or Flatpak to provide a standalone testcase.

**Comment by absidue** [comment]
Likely related to #43570, which would mean that your file chooser portal is too old.

**Comment by bermeitinger-b** [comment]
How would I check this?
```bash
$ /usr/lib/xdg-desktop-portal --version
xdg-desktop-portal 1.18.4

$ usr/lib/xdg-desktop-portal-gtk --version
xdg-desktop-portal-gtk 1.15.1
```

Or is it something else that I'm missing?

**Comment by bermeitinger-b** [comment]
When run with `--enable-logging=stderr --v=2`, I get a bit more output:
```
[2:0926/215407.125525:VERBOSE1:select_file_dialog_linux_portal.cc(293)] File chooser portal available: no
```

This can only happen from Chromium https://github.com/chromium/chromium/blob/ef09ea85d6fdb0f6002f0d7a986800c8b7834dc2/ui/shell_dialogs/select_file_dialog_linux_portal.cc#L291
when the version is `>= 4` (from your patch).

However, if I ask my dbus, the version of the FileChooser portal is 3 and not 4.

**Comment by bermeitinger-b** [comment]
Apparently, version 4 will be release with with the xdg-desktop-portal 1.20.x series (https://github.com/flatpak/xdg-desktop-portal/issues/1445#issuecomment-2379375660)

**Comment by codebytere** [comment]
That's correct - we intentionally chose to support a v4 minimum owing to broken support for `defaultPath` in save dialogs. If there are other options that guarantee support for that (added via https://github.com/flatpak/xdg-desktop-portal/commit/71165a52ea6ba164816131ea039ccea253fe103b) we're open to them, but we're not likely to change this outside the scope of that.

**Comment by bermeitinger-b** [comment]
Doesn't this break then all current usages of the file portals anywhere? What's the recommended process until 1.20.x is released and installed on a large basis?

**Comment by jarek-foksa** [comment]
From my testing this is also broken in Electron version 31.6.0, but not in versions 31.5.0 and 32.0.2

**Comment by ccdunder** [comment]
> Doesn't this break then all current usages of the file portals anywhere? What's the recommended process until 1.20.x is released and installed on a large basis?

Sure @bermeitinger-b, but look at it this way: for the next 1-2 years, users won't experience broken support for `defaultPath` in save dialogs because they won't be able to open any xdg portals whatsoever. Problem solved. /s

**Comment by Cknight70** [comment]
I would like to see a developer response considering the version of xdg-desktop-portal needed doesn't exist yet.

**Comment by deepak1556** [comment]
https://github.com/electron/electron/pull/44426 offers a way around this issue that can be adopted in the flaptak applications. @bermeitinger-b let us know if it helps.

**Comment by codebytere** [comment]
yes, @ccdunder @bermeitinger-b @Cknight70 we've heard your feedback and understand where you're coming from. The dev team has opened a PR linked above which allows conditionally addressing the concern from my original PR while not breaking portal usage for distributions without the latest version of `xdg-desktop-portal`.

**Comment by absidue** [comment]
As the original point of setting the version to 4 was to avoid the broken defaultPath behaviour, would it make sense to document the problems with version 3 in the documentation for that flag or log a warning at runtime.  That way if people set it to 3 and then complain about defaultPath being broken you can point them towards the docs or the logged warning.

**Comment by Erick555** [comment]
Since same flatpak needs to work on all distros then all electron flatpak apps would need to hardcode version 3 for many years ahead as currently released distros like ubuntu lts would never upgrade major xdg-portal version.

**Comment by RalfJung** [comment]
> The maintainer team has discussed and Deepak has opened a PR linked above which allows conditionally addressing the concern from my original PR while not breaking portal usage for distributions without the latest version of xdg-desktop-portal.

Thanks for that!

Just a small correction -- even distributions with the latest released version of xdg-desktop-portal are affected. A development pre-release is required for protocol version 4, and even that pre-release got tagged a month *after* you started depending on it.

Seems like a time machine has now also become a requirement to run Electron inside flatpak.

**Comment by Erick555** [comment]
That's right. ATM only 2 [distros](https://repology.org/project/xdg-desktop-portal/versions) in their development branch contain x-d-p 1.19  and that's unlikely to change until version 1.20 is released.

**Comment by codebytere** [comment]
Yes - we're also going to change the default back to 3 so that apps are responsible for setting the base to 4 if they critically care about the save dialog issue in Portals.

**Comment by A6GibKm** [comment]
I am not sure why there is a need to require a specific portal version. The version can be read at runtime and one can take advantage of its features if available or just not make use of those specific feature otherwise.

**Comment by czhang03** [comment]
I have noticed that the PR (https://github.com/electron/electron/pull/44426) only covers electron 34, which will require a month to be stable, and even longer for app to adopt it (vscode and freetube is still on 32, despite it has already been stable for around 1.5 month).

Given the breaking changes is promptly backported to all versions (even 30!) within a week (https://github.com/electron/electron/pull/43570, https://github.com/electron/electron/pull/43631, https://github.com/electron/electron/pull/43629, https://github.com/electron/electron/pull/43630, https://github.com/electron/electron/pull/43595), so I was wondering is there any particular blocker that will prevent the revert changes to be backported to older versions?

Thanks in advance!

**Comment by rijenkii** [comment]
Autobackport of #44426 to 32 and 33 has failed and nobody seems to want to manually backport them. Is that a tactic to force applications to use newer electron versions or just plain old "low priority"?

**Comment by codebytere** [comment]
@rijenkii December is Electron's Quiet Month - https://www.electronjs.org/blog/dec-quiet-period-24; availability is low at the moment. It'd be approved if someone external did it or after the quiet month ends and maintainer availability returns to normal.

**Comment by gmanskibiditoilet** [comment]
We think as of for now, the temporary fix for this problem as of today, is to ensure the application identifies your computer's filesystem by enabling file access, thus [Flatseal ](https://flathub.org/apps/com.github.tchx84.Flatseal)might be beneficial to solve the problem.

**Comment by codebytere** [comment]
Closed in https://github.com/electron/electron/pull/45193

**Comment by guihkx** [comment]
Just a reminder that this also affects 33.x, and there's a work-in-progress pull request for that here: https://github.com/electron/electron/pull/45180
